sources/packages/api/src/route-controllers/models/common.dto.ts

```diff
@@ -35,37 +36,42 @@ export class DynamicAPIOutDTO<T> {
  * Payload/Input DTO for address details.
  */
 export class AddressDetailsAPIInDTO {
-  @Allow()
+  @IsNotEmpty()
   addressLine1: string;
-  @Allow()
+  @IsOptional()
   addressLine2?: string;
-  @Allow()
+  @IsNotEmpty()
   city: string;
   // This property has the country irrespective of the selectedCountry and otherCountry,
   // so for API purpose take country from this property.
-  @Allow()
+  @IsNotEmpty()
   country: string;
   // This property has the postalCode irrespective of the selectedCountry and otherCountry,
   // so for API purpose take postalCode from this property.
-  @Allow()
+  @IsNotEmpty()
   postalCode: string;
-  @Allow()
+  // provinceState will only have value when selectedCountry is 'canada'.
+  @ValidateIf((value) => value.selectedCountry === COUNTRY_CANADA)
+  @IsNotEmpty()
   provinceState?: string;
   // This property will have canada postal code.
   // This property is for dry run validation, this is not saved in db.
-  @IsOptional()
+  @ValidateIf((value) => value.selectedCountry === COUNTRY_CANADA)
```

Comment:
Are we still running dry run for this? If not, please adjust the comments.

Reply:
this is the common interface, it's used in both dry run, and non-dry run controllers, as we discussed yesterday. i will adjust the commnets